src/module.c

```diff
@@ -7461,6 +7734,19 @@ RedisModuleBlockedClient *RM_BlockClient(RedisModuleCtx *ctx, RedisModuleCmdFunc
     return moduleBlockClient(ctx,reply_callback,timeout_callback,free_privdata,timeout_ms, NULL,0,NULL,0);
 }
 
+/* Block the current client for custom authentication in the background. If custom auth is not in
+ * progress, the API returns NULL. Otherwise, the client is blocked and the RM_BlockedClient is
+ * returned similar to the RM_BlockClient API. */
+RedisModuleBlockedClient *RM_BlockClientOnAuth(RedisModuleCtx *ctx, RedisModuleCustomAuthCallback reply_callback, void (*free_privdata)(RedisModuleCtx*,void*)) {
+    /* If the client is not in the middle of custom auth, return NULL. */
+    if (ctx && ctx->client && !(ctx->client->flags & CLIENT_CUSTOM_AUTH)) {
+        return NULL;
+    }
```

Comment:
I'm not sure we should silently return NULL on input validation. I would almost rather assert fail, since this means we're calling in an invalid context.

Reply:
Resolved in https://github.com/redis/redis/pull/11659/commits/a2cfa54fd955cdd84935377679b4abb55e917501